ICS_372_Group_Project/src/groupProject/VehicleJSONParser.java

```diff
@@ -0,0 +1,85 @@
+package groupProject;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.util.Map;
+import java.nio.file.Path;
+
+import org.codehaus.jackson.map.ObjectMapper;
+import org.json.simple.JSONArray;
+import org.json.simple.JSONObject;
+import org.json.simple.parser.JSONParser;
+import org.json.simple.parser.ParseException;
+
+/*
+ * The VehicleJSONParser class has one constructor 
+ * that takes in the file location of a JSON file as a String. 
+ * The file is parsed into a JSONArray.
+ * The class contains a method that creates a map from the JSONObject.
+ * Additionally, the JSONObject / JSONArray can be queried directly 
+ * to get it's contents.
+ */
+public class VehicleJSONParser {
+	
+	private Path file;
+	private String fileString;
+	private Object obj;
+	private JSONObject jo;
+	private JSONArray jsonArray = new JSONArray();
+	
+	VehicleJSONParser(String filePath) throws IOException, ParseException{
+		this.file = Path.of(filePath);
+		this.fileString = Files.readString(file);
+		this.obj = new JSONParser().parse(fileString);
+		this.jo = (JSONObject) obj;
+		this.jsonArray = (JSONArray) jo.get("car_inventory");
+	}
+
+	public JSONArray getJSONArray() {
+		return jsonArray;
+	}
+
+	/*
+	 * The buildMap method builds a map from the 
+	 * JSONObject in the JSONParser class. 
+	 * The map that is returned can be used to 
+	 * get vehicle and dealer information to build 
+	 * vehicle and dealer classes.
+	 */
+	public Map<String, Object> buildMap() {
```

Comment:
I thought this was changed earlier otherwise I would have written a comment. Do you think you could have this return Vehicles? However you choose, it just has to return the Vehicle objects in some form. 

I currently can't access that information without using ObjectMapper again which defeats the purpose of this method.